### PR TITLE
Multiple fixes (#67, #68, #71, #72, #73)

### LINKED ISSUE #67: Rouding of req/s is incorrect
We should have 2 decimals after ',' as if we have 0.5 req/s we show 0req/s

### LINKED ISSUE #68: Some metrics are incorrect
Comparing to Summary Report I have differences in:
- Error rate: 2.88% for SR vs 0% for plugin, error is on plugin
- Max:763ms vs 380ms for plugin, error is on plugin
- Percentiles are identical which is wrong

Screenshot 2023-04-08 at 15 43 51
Screenshot 2023-04-08 at 15 44 21

### LINKED ISSUE #71: Responses per second shows 0 response while it's not 0
Screenshot 2023-04-08 at 16 03 08

### LINKED ISSUE #72: Web dashboard improvements
- [x] Add unit (ms) for response times
- [x] Average response => Average response times
- [x] Max response => Max response times

### LINKED ISSUE #73: The 3 percentiles have the same value
Is it due to precision ? then should be documented

Screenshot 2023-04-08 at 16 06 16

Screenshot 2023-04-08 at 16 07 12

### PR DESCRIPTION
**Pull request by Liard-Thomas**
This part correctly displays the cumulated metrics of the test, instead of the last recorded period. These datas match the summary report values at the end of the test.
result
Threads value correspond to the max number of threads running at the same time for every periods.

### CODE REVIEW

**Comment by Liard-Thomas** [comment]
@pmouawad 
I unintentionally fixed the bug of the issue 'Responses per second shows 0 response while it's not 0 #71 '
-> It was a rounding issue where if you had a very low request/sec rate (something below 0.5), it would round it to 0.

I didn't find a way to reproduce the bug otherwise, so maybe it is not what caused the bug. I am saying that because you made 2 distinct issues. The first one is saying that the rouding of req/s is incorrect (#67), while the other show that you have 0 req/s on a test (#71). Correct me if i am wrong, but these 2 are the same issue in reality.

File: ulp-observability-front/src/main/front/src/app/components/ulp-observability-statistics/ulp-observability-statistics.component.ts [hunk 1]
```diff
@@ -99,14 +99,16 @@ export class UlpObservabilityStatisticsComponent implements OnChanges,OnInit {
             },
           };
 
-          ['avg','max','total','throughput'].forEach(type =>{
-            stats[type].data.value = this.datasets[type][samplerName][lastIndex].y;
+          ['avg','max','throughput'].forEach(type =>{
```
Comment (pmouawad) [inline ulp-observability-front/src/main/front/src/app/components/ulp-observability-statistics/ulp-observability-statistics.component.ts:109]: Use same approach also for total

File: back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/metric/ResponseResult.java [hunk 2]
```diff
@@ -47,13 +57,16 @@ public class ResponseResult {
      * @param allThreads Total thread count
      * @param samplerLabel Name of the sampler the sample belongs to
      */
```
Comment (pmouawad) [inline back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/metric/ResponseResult.java:61]: Javadocs missing

File: back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/registry/MicrometerRegistry.java [hunk 3]
```diff
@@ -49,6 +56,8 @@ public class MicrometerRegistry {
 	 */
 	private SampleLogger logger;
 	
+	private volatile Map<String, Pair<Long,Long>> startAndEndDatesOfSamplers = new ConcurrentHashMap<>();
```
Comment (pmouawad) [inline back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/registry/MicrometerRegistry.java]: volatile is useless

File: back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/registry/MicrometerRegistry.java [hunk 4]
```diff
@@ -124,32 +136,65 @@ public synchronized void addResponse(ResponseResult result) {
 			return;
 		}
 
+		// Create micrometer tags to feed with data
 		String threadTag = Util.makeMicrometerName(result.getThreadGroupLabel());
```
Comment (pmouawad) [inline back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/registry/MicrometerRegistry.java:143]: threadTag is not useful

File: back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/registry/MicrometerRegistry.java [hunk 5]
```diff
@@ -124,32 +136,65 @@ public synchronized void addResponse(ResponseResult result) {
...
-		this.registry.counter("count.threads", "sample", samplerTag).increment(
-				result.getGroupThreads() - (int) this.registry.counter("count.threads", "sample", threadTag).count());
-	
-		if(result.hasError()) {
-			this.registry.counter("count.error", "sample", threadTag).increment();
-			this.registry.counter("count.error", "sample", this.totalLabel).increment();
-			this.registry.counter("count.error", "sample", samplerTag).increment();
+		for(String microMeterTag : micrometerTags) {
+			// Save current period responses time
+			this.registry.summary("summary.response", "sample", microMeterTag).record(result.getResponseTime());
```
Comment (pmouawad) [inline back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/registry/MicrometerRegistry.java]: for clarity can you rename registry to intervalRegistry
and totalReg to summaryRegistry 

Thanks

File: back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/registry/MicrometerRegistry.java [hunk 6]
```diff
@@ -74,34 +83,37 @@ public MicrometerRegistry(
...
+			public DistributionStatisticConfig configure(
+					final Meter.Id id,
+					final DistributionStatisticConfig config
+					) {
+				if(id.getType().equals(
+						Type.DISTRIBUTION_SUMMARY
+						)) {
+					return DistributionStatisticConfig
+							.builder()
+							.percentilePrecision(
```
Comment (pmouawad) [inline back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/registry/MicrometerRegistry.java]: Shouldn't you adjust bufferLength and expiry ? 
I guess that also setting minimumExpectedValue and maximumExpectedValue can be useful

**Review by pmouawad — CHANGES_REQUESTED** [review]
Thanks for PR @Liard-Thomas , I let you fix the remarks

**Pull request merged**